### PR TITLE
feat(CodeArts): add a new resource codearts deploy group

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

add a new resource codearts deploy group

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- The field `is_proxy_mode` does not have response value.
- The Create\Update\Delete function need to add Header `"Content-Type": "application/json;charset=utf-8"`
- Even if the API response code is 200, the result may still be wrong, we need to parse the response body to check whether the request is success. 
- I have verified `CheckDeletedDiag`：
  - Step1: Use  terraform create a group resource.
  - Step2: Delete this group from console.
  - Step3: Run `terraform destroy` in terraform.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/codearts' TESTARGS='-run TestAccDeployGroup_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/codearts -v -run TestAccDeployGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDeployGroup_basic 
=== PAUSE TestAccDeployGroup_basic
=== CONT  TestAccDeployGroup_basic
--- PASS: TestAccDeployGroup_basic (14.98s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/codearts  15.028s
```

```
make testacc TEST='./huaweicloud/services/acceptance/codearts' TESTARGS='-run TestAccDeployGroup_resourcePoolId'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/codearts -v -run TestAccDeployGroup_resourcePoolId -timeout 360m -parallel 4 
=== RUN   TestAccDeployGroup_resourcePoolId 
=== PAUSE TestAccDeployGroup_resourcePoolId
=== CONT  TestAccDeployGroup_resourcePoolId
--- PASS: TestAccDeployGroup_resourcePoolId (21.19s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/codearts  21.245s
```

```
make testacc TEST='./huaweicloud/services/acceptance/codearts' TESTARGS='-run TestAccDeployGroup_errorCheck'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/codearts -v -run TestAccDeployGroup_errorCheck -timeout 360m -parallel 4 
=== RUN   TestAccDeployGroup_errorCheck 
=== PAUSE TestAccDeployGroup_errorCheck 
=== CONT  TestAccDeployGroup_errorCheck
--- PASS: TestAccDeployGroup_errorCheck (4.69s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/codearts  4.736s
```

### CODE REVIEW

File: docs/resources/codearts_deploy_group.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the group name. The name consists of 3 to 128 characters, including letters,
+  digits, chinese characters or `-_.` symbols.
+
+* `project_id` - (Required, String, ForceNew) Specifies the project ID for CodeArts service.
+
+  Changing this parameter will create a new resource.
+
+* `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid value are **windows** and **linux**.
```
Comment (Lance52259) [inline docs/resources/codearts_deploy_group.md]:
```suggestion
* `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid value is **windows** or **linux**.
```
or
```suggestion
* `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid values are **windows** and **linux**.
```

File: huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenDeployGroupCreateOrUpdateUser(resp interface{}, expression string) []interface{} {
+	curJson, err := jmespath.Search(expression, resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing %s from response= %#v, error: %s", expression, resp, err)
+		return []interface{}{}
```
Comment (Lance52259) [inline huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go]:
```suggestion
		return nil
```
how about this

File: huaweicloud/services/acceptance/codearts/resource_huaweicloud_codearts_deploy_group_test.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_codearts_deploy_group" "test" {
+  project_id       = huaweicloud_codearts_project.test.id
+  name             = "%s_update"
+  os_type          = "windows"
+  description      = "test description update"
+  is_proxy_mode    = 0
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/codearts/resource_huaweicloud_codearts_deploy_group_test.go]:
```suggestion
resource "huaweicloud_codearts_deploy_group" "test" {
  project_id    = huaweicloud_codearts_project.test.id
  name          = "%s_update"
  os_type       = "windows"
  description   = "test description update"
  is_proxy_mode = 0
}
```

File: docs/resources/codearts_deploy_group.md (hunk 1 above)
Comment (deer-hang) [reply docs/resources/codearts_deploy_group.md]: OK

File: huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go (hunk 2 above)
Comment (deer-hang) [reply huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go]: ok

File: huaweicloud/services/acceptance/codearts/resource_huaweicloud_codearts_deploy_group_test.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/acceptance/codearts/resource_huaweicloud_codearts_deploy_group_test.go]: OK

File: docs/resources/codearts_deploy_group.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
+---
+subcategory: "CodeArts"
```
Comment (ShiChangkuo) [inline docs/resources/codearts_deploy_group.md]: the subcategory should be **CodeArts Deploy**

File: huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+type deployErrorCode string
+
+const (
+	groupNotFound deployErrorCode = "Deploy.00021104"
```
Comment (ShiChangkuo) [inline huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go]: can we define the variable directly as string type?

File: huaweicloud/config/endpoints.go [hunk 6]
```diff
@@ -789,6 +789,12 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		Product: "ProjectMan",
 	},
 
+	"codearts_deploy": {
+		Name:    "codearts-deploy",
+		Version: "v2",
+		Product: "CodeArts",
```
Comment (ShiChangkuo) [inline huaweicloud/config/endpoints.go]:
```suggestion
		Product: "CodeArtsDeploy",
```

File: docs/resources/codearts_deploy_group.md (hunk 4 above)
Comment (deer-hang) [reply docs/resources/codearts_deploy_group.md]: OK

File: huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go (hunk 5 above)
Comment (deer-hang) [reply huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_group.go]: OK

File: huaweicloud/config/endpoints.go (hunk 6 above)
Comment (deer-hang) [reply huaweicloud/config/endpoints.go]: OK

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/codearts' TESTARGS='-run TestAccDeployGroup_basic'

==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/codearts -v -run TestAccDeployGroup_basic -timeout 360m -parallel 4 
=== RUN   TestAccDeployGroup_basic 
=== PAUSE TestAccDeployGroup_basic
=== CONT  TestAccDeployGroup_basic
--- PASS: TestAccDeployGroup_basic (17.79s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/codearts  17.840s
```

**Pull request merged**